### PR TITLE
Print timestamps with full precision for JSON

### PR DESCRIPTION
**Pull request by dominiklohmann**

### CODE REVIEW

File: libvast/vast/concept/printable/vast/json.hpp [hunk 1]
```diff
@@ -122,6 +122,11 @@ struct json_printer : printer<json_printer<TreePolicy, Indent, Padding>> {
       return (*this)(x.string());
     }
 
+    bool operator()(const view<time>& x) {
+      static auto p = '"' << data_view_printer{} << '"';
```
Comment (mavam) [inline libvast/vast/concept/printable/vast/json.hpp:126]: Why not `printers::time_point`?

Comment (dominiklohmann) [reply libvast/vast/concept/printable/vast/json.hpp:126]: Fair enough. Changed it.

**Comment by dominiklohmann** [comment]
> I'd like to see some sort of test that captures this change.

I just forgot to update the tests, they failed in CI. They're now fixed.

**Comment by dominiklohmann** [comment]
I rebased to fix the merge conflict in the changelog.

**Pull request merged**